### ISSUE TITLE
`jsdoc/no-undefined-types` incorrect behavior together with the TS `is` operator

### ISSUE THREAD

**Issue by what1s1ove** [gajus/eslint-plugin-jsdoc#1591]
First of all, thank you for your work on this plugin. I currently have the following issue. Currently, the jsdoc/no-undefined-types rule behaves strangely with [the TypeScript `is` operator](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

## Expected behavior
```js
/**
 * @template {unknown} T
 * @param {unknown} value
 * @param {...T} validValues
 * @returns {value is T}
 */
const checkIsOnOf = (value, ...validValues) => {
	return validValues.includes(value);
};
```

With no `jsdoc/no-undefined-types` error.

## Actual behavior
The code above will fail with the error: "5:1  error  The type 'value' is undefined  jsdoc/no-undefined-types". Please note that the syntax is fully valid TypeScript, and TypeScript handles it just fine.

## Environment

- Node version: 24.11.0
- ESLint version: 9.39.0
- `eslint-plugin-jsdoc` version: 61.1.11
PS. I upgraded from `eslint-plugin-jsdoc` version 46.9.0, and this issue did not exist in that version. Although I know that, in general, `no-undefined-types` is a problematic rule 🫠

**Comment by what1s1ove** [comment]
Thanks @brettz9
